modules/user/lib/user-sync-aura-balance.service.ts

```diff
@@ -1,60 +1,106 @@
+import { Chain, PrismaPoolStakingType } from '@prisma/client';
 import ERC20Abi from '../../web3/abi/ERC20.json';
 import { AuraSubgraphService } from '../../sources/subgraphs/aura/aura.service';
 import { formatEther, hexToBigInt } from 'viem';
-import config from '../../../config';
 import { getViemClient } from '../../sources/viem-client';
+import config from '../../../config';
+import { AccountSchemaFragment } from '../../sources/subgraphs/aura/generated/aura-subgraph-types';
 
 export class UserSyncAuraBalanceService implements UserStakedBalanceService {
     public async initStakedBalances(stakingTypes: PrismaPoolStakingType[], chain: Chain): Promise<void> {
         if (!stakingTypes.includes('AURA')) {
             return;
         }
 
-        const auraSubgraphService = new AuraSubgraphService(config[chain].subgraphs.aura!);
         const viemClient = getViemClient(chain);
         const blockNumber = await viemClient.getBlockNumber();
-        const auraGauges = await auraSubgraphService.getAllPools([chain]);
-        const accounts = await auraSubgraphService.getAllUsers();
 
-        const pools = await prisma.prismaPool.findMany({
-            select: { id: true, address: true, staking: true },
-            where: {
-                chain: chain,
-                staking: {
-                    some: { aura: { auraPoolAddress: { in: auraGauges.map((auraGauge) => auraGauge.address) } } },
-                },
-            },
+        let accounts: AccountSchemaFragment[] = [];
+        try {
+            const auraSubgraphService = new AuraSubgraphService(config[chain].subgraphs.aura!);
+            accounts = await auraSubgraphService.getAllUsers();
+        } catch (e) {
+            return;
+        }
+
+        // Get AURA pools - used to deal with staking ID DB constraint
+        const stakings = await prisma.prismaPoolStaking.findMany({
+            select: { id: true, poolId: true },
+            where: { type: 'AURA', chain },
+        });
+
+        const dbBalances = await prisma.prismaUserStakedBalance.findMany({
+            where: { chain: chain, stakingId: { in: stakings.map((staking) => staking.id) } },
         });
 
+        const dbBalancesMap = _.keyBy(dbBalances, (dbBalance) => `${dbBalance.stakingId}-${dbBalance.userAddress}`);
+
         const operations: any[] = [];
+        let i = 0;
         for (const account of accounts) {
             for (const poolAccount of account.poolAccounts) {
                 if (poolAccount.pool.chainId === config[chain].chain.id) {
-                    const pool = pools.find((pool) => pool.address === poolAccount.pool.lpToken.address);
-                    if (!pool) {
+                    let staking = stakings.find((s) => s.id === poolAccount.pool.address);
+
+                    // Add new staking
+                    if (!staking) {
+                        const pool = await prisma.prismaPool.findFirst({
```

Comment:
dont think you actually need to query the DB again here, you can get the pool ID from `stakings`

Reply:
this is handling the case when staking isn't synced and the AURA pool is still missing in the DB. it's just adding it here instead of failing, or skipping